extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3InputDataConfig.java

```diff
@@ -29,16 +32,33 @@
  */
 public class S3InputDataConfig
 {
+  @VisibleForTesting
+  static final int MAX_LISTING_LENGTH_MIN = 1;
+
+  /**
+   * AWS S3 only allows deleting 1000 elements at a time:
+   * https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/DeleteObjectsRequest.html
```

Comment:
Does this only affect deleting objects? If the API allows you to list more than 1000 objects, I think a better fix would be in `S3Utils#deleteObjectsInPath` to check if `keysToDelete.size() >= Math.min(1000, config.getMaxListingLength())`

You could even extract `Math.min(1000, config.getMaxListingLength())` to a JsonIgnore function in this config file. This way there's a clear distinction between listing many files and the number of files you're allowed to delete in a single request

Reply:
It's not super clear whether or not list objects API is limited on the server side to a max of 1000, From https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html#API_ListObjects_RequestSyntax

>Sets the maximum number of keys returned in the response. By default the API returns up to 1,000 key names. The response might contain fewer keys but will never contain more.

Reply:
Yeah I think we should leave both listing and deleting to max of 1000 keys. I can add the link to this documentation in the javadocs as well.

Reply:
This seems ok to me, 1k is probably enough to list even if it isn't the true maximum limit the API can handle.